### PR TITLE
Exclude water, but include water-solute hydrogen bonding

### PR DESCRIPTION
**Pull request by ahy3nz**
Addressing issue #1269. Main idea is to preserve the original logic with exclude_water, but add a scenario in which waters are largely excluded, but still look at water-solute or solute-water hydrogen bonding. This was implemented mainly by changing what constitutes the donor or acceptor set to now account for water. Comments/improvements would be greatly appreciated

### CODE REVIEW

File: mdtraj/geometry/tests/test_hbonds.py [hunk 1]
```diff
@@ -145,3 +145,23 @@ def test_wernet_nilsson_1():
...
+    hbonds1 = md.baker_hubbard(t, exclude_water=True, include_water_solute=True) 
+    assert np.shape(hbonds1)[0] > 0 
+
+    # No water, no water-solute
+    hbonds2 = md.baker_hubbard(t, exclude_water=True, include_water_solute=False)
+    assert np.shape(hbonds2)[0] == 0
+
+    # Yes water, yes water-solute (water-solute is redundant)
+    hbonds3 = md.baker_hubbard(t, exclude_water=False, include_water_solute=True)
+    assert np.shape(hbonds3)[0] > 0
```
Comment (ctk3b) [inline mdtraj/geometry/tests/test_hbonds.py]: Could probably also assert that hbonds3 contains more than hbonds1?

Comment (ahy3nz) [inline mdtraj/geometry/tests/test_hbonds.py]: Good point, adding that

File: mdtraj/geometry/tests/test_hbonds.py [hunk 2]
```diff
@@ -145,3 +145,24 @@ def test_wernet_nilsson_1():
             # to make sure the criterion is giving back totally implausible stuff
             if len(hbonds) > 0:
                 assert np.all(md.compute_distances(t[frame], hbonds[:, [0,2]]) < 0.5)
+def test_water_solute():
```
Comment (tcmoore3) [inline mdtraj/geometry/tests/test_hbonds.py]: Should you test or donor-acceptor selections? I assume you can manually calculate how long the donors and acceptors lists should be.

Comment (ahy3nz) [inline mdtraj/geometry/tests/test_hbonds.py]: For example, for water-solute, verify the number of water donors and number of solute acceptors? And then do the same thing for solute-water?

Comment (tcmoore3) [inline mdtraj/geometry/tests/test_hbonds.py]: Yeah.

**Comment by mpharrigan** [comment]
Another note: try to maintain backwards compatibility by adding new optional arguments to the end so the api doesn't change w.r.t. positional args

**Pull request closed without merging**